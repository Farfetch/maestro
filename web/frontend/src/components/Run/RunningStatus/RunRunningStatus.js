import {
  Button,
  Col,
  PageHeader,
  Popconfirm,
  Row,
  Space,
  Tabs,
  Typography
} from "antd";
import { useCallback, useState } from "react";
import { useNavigate } from "react-router-dom";

import { restartRun } from "../../../lib/api/endpoints/run";
import { runStatus as runStatusModel } from "../../../lib/api/models";
import { historyUrl, testSingleUrl } from "../../../lib/routes";
import Breadcrumb from "../../layout/Breadcrumb";
import RunStatusTag from "../../tag/RunStatusTag";
import RunAnalyticCharts from "./AnalyticCharts";
import DownloadMetricsButton from "./DownloadMetricsButton";
import RunEditableLabelsGroup from "./EditableLabelsGroup";
import RunEndpointCharts from "./EndpointCharts";
import MoreButtonsMenu from "./MoreButtonsMenu";
import RunNotesInput from "./NotesInput";
import RunRunningTime from "./RunningTime";
import StopExecutionButton from "./StopExecutionButton";
import RunSummaryTable from "./SummaryTable";

const RunRunningStatus = ({ run }) => {
  const navigate = useNavigate();
  const [isRunMetricsAvailable, setIsRunMetricsAvailable] = useState(false);

  const routes = [
    {
      path: historyUrl,
      breadcrumbName: "History"
    },

    {
      path: testSingleUrl(run.runConfigurationId),
      breadcrumbName: "Configuration"
    },
    {
      path: "index",
      breadcrumbName: "Run"
    }
  ];

  const onRestart = useCallback(() => {
    restartRun(run.id);
  }, [run]);

  const isTimerAvailable =
    (run.runStatus === runStatusModel.RUNNING && isRunMetricsAvailable) ||
    run.runStatus === runStatusModel.STOPPED ||
    run.runStatus === runStatusModel.FINISHED;

  const getButtonsByStatus = (runStatus) => {
    switch (runStatus) {
      case runStatusModel.STOPPED:
      case runStatusModel.FINISHED:
        return [
          <Popconfirm
            key="restart"
            placement="left"
            title={
              <>
                <p>Are you sure you want to restart test?</p>
                <p>All metrics would be deleted. </p>
              </>
            }
            okText="Yes"
            cancelText="No"
            onConfirm={onRestart}
          >
            <Button type="secondary">Restart</Button>
          </Popconfirm>,
          <DownloadMetricsButton runId={run.id} key="downloadMetrics" />,
          <MoreButtonsMenu runId={run.id} key="moreButtonsMenu" />
        ];
      default:
        if (!isRunMetricsAvailable) return [];
        return [<StopExecutionButton runId={run.id} key="stopExecution" />];
    }
  };

  return (
    <PageHeader
      ghost={false}
      onBack={() => navigate(-1)}
      title={run.title}
      subTitle={
        <RunStatusTag
          runStatus={run.runStatus}
          key={`tag-${run.id}-${run.runStatus}`}
        />
      }
      extra={getButtonsByStatus(run.runStatus)}
      breadcrumb={{
        routes,
        itemRender: (route, params, routesToRender) => (
          <Breadcrumb route={route} routes={routesToRender} />
        )
      }}
      tags={<RunEditableLabelsGroup runId={run.id} defaultValue={run.labels} />}
    >
      <Row gutter={[0, 0]} type="flex" align="middle" justify="end">
        <Col span={16}>
          <Space direction="vertical" size={0}>
            <RunNotesInput
              runId={run.id}
              defaultValue={run.notes || "Add running test execution notes..."}
            />

            <Space size="large">
              <div>
                <Typography.Text type="secondary">Started at: </Typography.Text>
                {run.startedAt ? run.startedAt.format("L HH:mm:ss") : null}
              </div>
              <div>
                <Typography.Text type="secondary">Agents: </Typography.Text>
                {run.agentIds.length}
              </div>
            </Space>
          </Space>
        </Col>
        <Col span={8} style={{ textAlign: "right" }}>
          {isTimerAvailable ? (
            <>
              <RunRunningTime
                startedAt={run.startedAt}
                finishedAt={
                  run.runStatus !== runStatusModel.RUNNING
                    ? run.finishedAt
                    : false
                }
              />
            </>
          ) : null}
        </Col>

        <Col span={24}>
          <Tabs defaultActiveKey="analytics">
            <Tabs.TabPane tab="Overview" key="overview">
              <RunAnalyticCharts
                run={run}
                loadProfile={run.loadProfile}
                isRunMetricsAvailable={isRunMetricsAvailable}
                setIsRunMetricsAvailable={setIsRunMetricsAvailable}
              />
            </Tabs.TabPane>
            <Tabs.TabPane
              tab="Summary"
              key="summary"
              disabled={!isRunMetricsAvailable}
            >
              <RunSummaryTable runId={run.id} />
            </Tabs.TabPane>
            <Tabs.TabPane
              tab="Endpoints"
              key="endpoints"
              disabled={!isRunMetricsAvailable}
            >
              <RunEndpointCharts run={run} />
            </Tabs.TabPane>
          </Tabs>
        </Col>
      </Row>
    </PageHeader>
  );
};
export default RunRunningStatus;
