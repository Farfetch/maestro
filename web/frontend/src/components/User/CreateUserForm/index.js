import CreateUserForm from "./CreateUserForm";

export default CreateUserForm;
