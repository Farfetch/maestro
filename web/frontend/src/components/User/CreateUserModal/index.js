import CreateUserModal from "./CreateUserModal";

export default CreateUserModal;
