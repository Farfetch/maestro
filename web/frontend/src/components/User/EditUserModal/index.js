import EditUserModal from "./EditUserModal";

export default EditUserModal;
