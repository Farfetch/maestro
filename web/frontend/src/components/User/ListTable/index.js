import UserListTable from "./UserListTable";

export default UserListTable;
