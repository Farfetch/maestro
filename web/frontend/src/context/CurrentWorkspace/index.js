export {
  CurrentWorkspaceContext,
  CurrentWorkspaceContextProvider
} from "./CurrentWorkspace";
