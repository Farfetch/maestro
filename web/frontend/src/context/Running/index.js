export { RunningContext, RunningContextProvider } from "./RunningContext";
