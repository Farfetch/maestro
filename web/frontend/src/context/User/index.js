export { UserContext, UserContextProvider } from "./UserContext";
