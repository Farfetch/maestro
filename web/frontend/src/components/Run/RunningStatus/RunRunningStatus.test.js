import { render } from "@testing-library/react";
import moment from "moment";
import React from "react";
import { MemoryRouter } from "react-router-dom";

import { UserContext } from "../../../context/User";
import RunRunningStatus from "./RunRunningStatus";

describe("RunRunningStatus", () => {
  const mockUserContext = {
    user: {
      name: "Test User"
    }
  };

  const mockRun = {
    id: "123",
    runConfigurationId: "456",
    title: "Test Run",
    runStatus: "RUNNING",
    startedAt: moment("2018-05-24T13:48:04.313000"),
    finishedAt: null, // Set to a date if the runStatus is not "RUNNING"
    agentIds: ["agent-1", "agent-2"],
    agentsNames: [
      {
        id: "agent-1",
        ip: "192.168.1.1",
        name: "Agent 1"
      },
      {
        id: "agent-2",
        ip: "192.168.1.2",
        name: "Agent 2"
      }
    ],
    labels: ["Label1", "Label2"],
    notes: "Test notes",
    loadProfile: {},
    isLoadProfileEnabled: true
  };

  test("should render RunRunningStatus component", () => {
    const rendered = render(
      <UserContext.Provider value={mockUserContext}>
        <MemoryRouter>
          <RunRunningStatus run={mockRun} />
        </MemoryRouter>
      </UserContext.Provider>
    );
    const component = rendered.container;
    expect(component.outerHTML).toMatchSnapshot();
  });
});
