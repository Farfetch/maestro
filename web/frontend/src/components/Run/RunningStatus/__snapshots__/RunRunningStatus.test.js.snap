// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RunRunningStatus should render RunRunningStatus component 1`] = `"<div><div class=\\"ant-page-header has-breadcrumb\\"><div class=\\"ant-breadcrumb\\"><span><span class=\\"ant-breadcrumb-link\\"><a href=\\"/history\\">History</a></span><span class=\\"ant-breadcrumb-separator\\">/</span></span><span><span class=\\"ant-breadcrumb-link\\"><a href=\\"/test/456\\">Configuration</a></span><span class=\\"ant-breadcrumb-separator\\">/</span></span><span><span class=\\"ant-breadcrumb-link\\"><span>Run</span></span><span class=\\"ant-breadcrumb-separator\\">/</span></span></div><div class=\\"ant-page-header-heading\\"><div class=\\"ant-page-header-heading-left\\"><div class=\\"ant-page-header-back\\"><div role=\\"button\\" tabindex=\\"0\\" class=\\"ant-page-header-back-button\\" aria-label=\\"Back\\" style=\\"border: 0px; background: transparent; padding: 0px; line-height: inherit; display: inline-block;\\"><span role=\\"img\\" aria-label=\\"arrow-left\\" class=\\"anticon anticon-arrow-left\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"arrow-left\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M872 474H286.9l350.2-304c5.6-4.9 2.2-14-5.2-14h-88.5c-3.9 0-7.6 1.4-10.5 3.9L155 487.8a31.96 31.96 0 000 48.3L535.1 866c1.5 1.3 3.3 2 5.2 2h91.5c7.4 0 10.8-9.2 5.2-14L286.9 550H872c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z\\"></path></svg></span></div></div><span class=\\"ant-page-header-heading-title\\"><div style=\\"width: 550px;\\"><div class=\\"ant-typography ant-typography-primary ant-typography-ellipsis ant-typography-single-line\\" aria-label=\\"Test Run\\" style=\\"padding-left: 14px; max-width: 500px; padding-top: 18px;\\">Test Run<div role=\\"button\\" tabindex=\\"0\\" class=\\"ant-typography-edit\\" aria-label=\\"Edit\\" style=\\"border: 0px; background: transparent; padding: 0px; line-height: inherit; display: inline-block;\\"><span role=\\"button\\" aria-label=\\"edit\\" class=\\"anticon anticon-edit\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"edit\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z\\"></path></svg></span></div><span aria-hidden=\\"true\\" style=\\"position: fixed; display: block; left: 0px; top: 0px; z-index: -9999; visibility: hidden; pointer-events: none; word-break: keep-all; white-space: nowrap;\\">lg</span><span aria-hidden=\\"true\\" style=\\"position: fixed; display: block; left: 0px; top: 0px; z-index: -9999; visibility: hidden; pointer-events: none; width: 0px; white-space: normal; margin: 0px; padding: 0px;\\"><span aria-hidden=\\"true\\">...</span><div role=\\"button\\" tabindex=\\"0\\" class=\\"ant-typography-edit\\" aria-label=\\"Edit\\" style=\\"border: 0px; background: transparent; padding: 0px; line-height: inherit; display: inline-block;\\"><span role=\\"button\\" aria-label=\\"edit\\" class=\\"anticon anticon-edit\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"edit\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z\\"></path></svg></span></div></span></div></div></span><span class=\\"ant-page-header-heading-sub-title\\"><span class=\\"ant-tag ant-tag-processing\\"><span role=\\"img\\" aria-label=\\"sync\\" class=\\"anticon anticon-sync anticon-spin\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"sync\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M168 504.2c1-43.7 10-86.1 26.9-126 17.3-41 42.1-77.7 73.7-109.4S337 212.3 378 195c42.4-17.9 87.4-27 133.9-27s91.5 9.1 133.8 27A341.5 341.5 0 01755 268.8c9.9 9.9 19.2 20.4 27.8 31.4l-60.2 47a8 8 0 003 14.1l175.7 43c5 1.2 9.9-2.6 9.9-7.7l.8-180.9c0-6.7-7.7-10.5-12.9-6.3l-56.4 44.1C765.8 155.1 646.2 92 511.8 92 282.7 92 96.3 275.6 92 503.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8zm756 7.8h-60c-4.4 0-7.9 3.5-8 7.8-1 43.7-10 86.1-26.9 126-17.3 41-42.1 77.8-73.7 109.4A342.45 342.45 0 01512.1 856a342.24 342.24 0 01-243.2-100.8c-9.9-9.9-19.2-20.4-27.8-31.4l60.2-47a8 8 0 00-3-14.1l-175.7-43c-5-1.2-9.9 2.6-9.9 7.7l-.7 181c0 6.7 7.7 10.5 12.9 6.3l56.4-44.1C258.2 868.9 377.8 932 512.2 932c229.2 0 415.5-183.7 419.8-411.8a8 8 0 00-8-8.2z\\"></path></svg></span><span>RUNNING</span></span></span><span class=\\"ant-page-header-heading-tags\\"><span class=\\"ant-tag\\"><span>Label1</span><span role=\\"img\\" aria-label=\\"close\\" tabindex=\\"-1\\" class=\\"anticon anticon-close ant-tag-close-icon\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"close\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z\\"></path></svg></span></span><span class=\\"ant-tag\\"><span>Label2</span><span role=\\"img\\" aria-label=\\"close\\" tabindex=\\"-1\\" class=\\"anticon anticon-close ant-tag-close-icon\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"close\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z\\"></path></svg></span></span><button type=\\"button\\" class=\\"ant-btn ant-btn-dashed ant-btn-sm\\"><span role=\\"img\\" aria-label=\\"plus\\" class=\\"anticon anticon-plus\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"plus\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><defs><style></style></defs><path d=\\"M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z\\"></path><path d=\\"M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z\\"></path></svg></span><span>New label</span></button></span></div><span class=\\"ant-page-header-heading-extra\\"></span></div><div class=\\"ant-page-header-content\\"><div type=\\"flex\\" class=\\"ant-row ant-row-end ant-row-middle\\"><div class=\\"ant-col ant-col-16\\"><div class=\\"ant-space ant-space-vertical\\"><div class=\\"ant-space-item\\" style=\\"margin-bottom: 0px;\\"><div class=\\"ant-typography ant-typography-secondary\\">Test notes<div role=\\"button\\" tabindex=\\"0\\" class=\\"ant-typography-edit\\" aria-label=\\"Edit\\" style=\\"border: 0px; background: transparent; padding: 0px; line-height: inherit; display: inline-block;\\"><span role=\\"button\\" aria-label=\\"edit\\" class=\\"anticon anticon-edit\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"edit\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z\\"></path></svg></span></div></div></div><div class=\\"ant-space-item\\"><div class=\\"ant-space ant-space-horizontal ant-space-align-center\\"><div class=\\"ant-space-item\\" style=\\"margin-right: 24px;\\"><div><span class=\\"ant-typography ant-typography-secondary\\">Started at: </span>05/24/2018 13:48:04</div></div><div class=\\"ant-space-item\\"><div><span class=\\"ant-typography ant-typography-secondary\\">Agents: </span>2</div></div></div></div></div></div><div style=\\"text-align: right;\\" class=\\"ant-col ant-col-8\\"></div><div style=\\"flex: 1; margin-top: 10px;\\" class=\\"ant-col\\"><div class=\\"ant-row ant-row-center ant-row-middle\\" style=\\"align-self: center;\\"><div class=\\"ant-spin ant-spin-spinning\\"><span role=\\"img\\" aria-label=\\"loading\\" style=\\"font-size: 64px;\\" class=\\"anticon anticon-loading anticon-spin ant-spin-dot\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"loading\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z\\"></path></svg></span></div></div></div><div class=\\"ant-col ant-col-24\\"><div class=\\"ant-tabs ant-tabs-top\\"><div role=\\"tablist\\" class=\\"ant-tabs-nav\\"><div class=\\"ant-tabs-nav-wrap\\"><div class=\\"ant-tabs-nav-list\\" style=\\"transform: translate(0px, 0px);\\"><div class=\\"ant-tabs-tab ant-tabs-tab-active\\"><div role=\\"tab\\" aria-selected=\\"true\\" class=\\"ant-tabs-tab-btn\\" tabindex=\\"0\\" id=\\"rc-tabs-test-tab-overview\\" aria-controls=\\"rc-tabs-test-panel-overview\\">Overview</div></div><div class=\\"ant-tabs-tab ant-tabs-tab-disabled\\"><div role=\\"tab\\" aria-selected=\\"false\\" class=\\"ant-tabs-tab-btn\\" aria-disabled=\\"true\\" id=\\"rc-tabs-test-tab-summary\\" aria-controls=\\"rc-tabs-test-panel-summary\\">Summary</div></div><div class=\\"ant-tabs-tab ant-tabs-tab-disabled\\"><div role=\\"tab\\" aria-selected=\\"false\\" class=\\"ant-tabs-tab-btn\\" aria-disabled=\\"true\\" id=\\"rc-tabs-test-tab-responseCodes\\" aria-controls=\\"rc-tabs-test-panel-responseCodes\\">Errors</div></div><div class=\\"ant-tabs-tab ant-tabs-tab-disabled\\"><div role=\\"tab\\" aria-selected=\\"false\\" class=\\"ant-tabs-tab-btn\\" aria-disabled=\\"true\\" id=\\"rc-tabs-test-tab-endpoints\\" aria-controls=\\"rc-tabs-test-panel-endpoints\\">Endpoints</div></div><div class=\\"ant-tabs-tab\\"><div role=\\"tab\\" aria-selected=\\"false\\" class=\\"ant-tabs-tab-btn\\" tabindex=\\"0\\" id=\\"rc-tabs-test-tab-initialConfiguration\\" aria-controls=\\"rc-tabs-test-panel-initialConfiguration\\">Initial Configuration</div></div><div class=\\"ant-tabs-ink-bar ant-tabs-ink-bar-animated\\"></div></div></div><div class=\\"ant-tabs-nav-operations ant-tabs-nav-operations-hidden\\"><button type=\\"button\\" class=\\"ant-tabs-nav-more\\" style=\\"visibility: hidden; order: 1;\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" aria-haspopup=\\"listbox\\" aria-controls=\\"rc-tabs-test-more-popup\\" id=\\"rc-tabs-test-more\\" aria-expanded=\\"false\\"><span role=\\"img\\" aria-label=\\"ellipsis\\" class=\\"anticon anticon-ellipsis\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"ellipsis\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z\\"></path></svg></span></button></div></div><div class=\\"ant-tabs-content-holder\\"><div class=\\"ant-tabs-content ant-tabs-content-top\\"><div role=\\"tabpanel\\" tabindex=\\"0\\" aria-hidden=\\"false\\" class=\\"ant-tabs-tabpane ant-tabs-tabpane-active\\" id=\\"rc-tabs-test-panel-overview\\" aria-labelledby=\\"rc-tabs-test-tab-overview\\"><div class=\\"ant-row ant-row-center ant-row-middle\\" style=\\"align-self: center;\\"><div class=\\"ant-spin ant-spin-spinning\\"><span role=\\"img\\" aria-label=\\"loading\\" style=\\"font-size: 64px;\\" class=\\"anticon anticon-loading anticon-spin ant-spin-dot\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"loading\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z\\"></path></svg></span></div></div></div><div role=\\"tabpanel\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"display: none;\\" class=\\"ant-tabs-tabpane\\" id=\\"rc-tabs-test-panel-summary\\" aria-labelledby=\\"rc-tabs-test-tab-summary\\"></div><div role=\\"tabpanel\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"display: none;\\" class=\\"ant-tabs-tabpane\\" id=\\"rc-tabs-test-panel-responseCodes\\" aria-labelledby=\\"rc-tabs-test-tab-responseCodes\\"></div><div role=\\"tabpanel\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"display: none;\\" class=\\"ant-tabs-tabpane\\" id=\\"rc-tabs-test-panel-endpoints\\" aria-labelledby=\\"rc-tabs-test-tab-endpoints\\"></div><div role=\\"tabpanel\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"display: none;\\" class=\\"ant-tabs-tabpane\\" id=\\"rc-tabs-test-panel-initialConfiguration\\" aria-labelledby=\\"rc-tabs-test-tab-initialConfiguration\\"></div></div></div></div></div></div></div></div></div>"`;
